test/testdata/rewriter/generic_module_function.rb

```diff
@@ -8,14 +8,14 @@
 # generic classes. This shouldn't be a problem with the DSL-ified
 # version.
 
-module M
+class M
   extend T::Sig
   extend T::Helpers
   extend T::Generic
 
-  A = type_member(:in)
+  A = type_member
 
-  interface!
+  abstract!
```

Comment:
I changed this test because we were using `module_function` in an interface: `module_function` creates private methods, and interfaces disallow private methods. Our `module_function` rewriter changes

```ruby
module_function def foo; end
```
into something like
```ruby
def foo; end
def self.foo; end
private :foo 
```
which is why this was allowed before, but now that the `private :foo` declaration actually means something this was raising an error.